src/test/scala/edu/ie3/simona/agent/grid/DBFSMockGridAgents.scala

```diff
@@ -40,36 +41,41 @@ trait DBFSMockGridAgents extends UnitSpec {
       : squants.electro.ElectricPotential = Volts(1e-6)
 
   sealed trait GAActorAndModel {
-    val gaProbe: TestProbe
+    val gaProbe: TestProbe[GridAgentMessage]
     val nodeUuids: Seq[UUID]
 
-    def ref: ActorRef = gaProbe.ref
+    def ref: ActorRef[GridAgentMessage] = gaProbe.ref
   }
 
   final case class InferiorGA(
-      override val gaProbe: TestProbe,
+      override val gaProbe: TestProbe[GridAgentMessage],
       override val nodeUuids: Seq[UUID],
   ) extends GAActorAndModel {
 
-    def expectGridPowerRequest(): ActorRef = {
-      gaProbe
-        .expectMsgType[RequestGridPowerMessage]
-        .nodeUuids should contain allElementsOf nodeUuids
+    def expectGridPowerRequest(): ActorRef[GridAgentMessage] = {
+      val message = gaProbe.expectMessageType[GridAgentMessage]
 
-      gaProbe.lastSender
+      message match {
+        case PMAdapter(requestGridPowerMessage: RequestGridPowerMessage) =>
+          requestGridPowerMessage.nodeUuids should contain allElementsOf nodeUuids
+
+          requestGridPowerMessage.sender
+      }
     }
 
     def expectSlackVoltageProvision(
         expectedSweepNo: Int,
         expectedExchangedVoltages: Seq[ExchangeVoltage],
     ): Unit = {
-      inside(gaProbe.expectMsgType[ProvideSlackVoltageMessage]) {
-        case ProvideSlackVoltageMessage(sweepNo, exchangedVoltages) =>
-          sweepNo shouldBe expectedSweepNo
+      val message = gaProbe.expectMessageType[GridAgentMessage]
```

Comment:
Minor point, but you could "inline" this variable as well (i.e. match on the return value directly)